Mathlib/Tactic/Autolabels/EnvExtension.lean

```diff
@@ -0,0 +1,248 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+
+/-!
+# Automatic labelling of PRs
+
+This file contains the basic definitions, the environment extension and the API commands
+to automatically assign a GitHub label to a PR.
+
+The heuristic used to assign a label is to assign labels to certain folder inside `Mathlib/`
+and assign a label to a PR if the PR modifies files within the corresponding folders.
+
+Ultimately, this produces a `HashMap String String` that takes a string representing a path
+to a modified file and returns the corresponding GitHub label.
+
+Since the labels are independent of the modified file, we define a `PersistentEnvExtension`
+that keeps track of an array of `Label`s.
+
+## The `Label` structure
+
+A `Label` is a structure containing
+* the GitHub label (as a `String` -- `Label.label`);
```

Comment:
What does `--` signify here? I don't understand.

Reply:
Maybe the following notation is more intuitive?

```
* `Label.label : String`; the Github label
```

Reply:
Jon's interpretation is indeed what I had in mind, but it looks like it was confusing.  I simply remove `-- ...`, since the source code is right below anyway.